_posts/2024-05-27-nanoarrow-0.5.0-release.md

```diff
@@ -0,0 +1,317 @@
+
+```python
+na.c_schema(pa.decimal128(10, 3))
+#> <nanoarrow.c_schema.CSchema decimal128(10, 3)>
+#> - format: 'd:10,3'
+#> - name: ''
+#> - flags: 2
+#> - metadata: NULL
+#> - dictionary: NULL
+#> - children[0]:
+
+na.c_array(["one", "two", "three", None], na.string())
+#> <nanoarrow.c_array.CArray string>
+#> - length: 4
+#> - offset: 0
+#> - null_count: 1
+#> - buffers: (4754305168, 4754307808, 4754310464)
+#> - dictionary: NULL
+#> - children[0]:
+```
+
+All nanoarrow type/array-like objects implement the
+[Arrow PyCapsule interface](https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html)
+for both producing and consuming and are zero-copy interchangeable with `pyarrow`
+objects in many cases:
+
+```python
+import pyarrow as pa
+
+pa.field(na.int32())
+#> pyarrow.Field<: int32>
+
+na.Schema(pa.string())
+#> <Schema> string
+
+pa.array(na.Array([4, 5, 6], na.int32()))
+#> <pyarrow.lib.Int32Array object at 0x11b552500>
+#> [
+#>   4,
+#>   5,
+#>   6
+#> ]
+
+na.Array(pa.array([10, 11, 12]))
+#> nanoarrow.Array<int64>[3]
+#> 10
+#> 11
+#> 12
+```
+
+For a more detailed tour of the nanoarrow Python bindings, see the
+[Getting started in Python guide](https://arrow.apache.org/nanoarrow/latest/getting-started/python.html) and the
+[Python API reference](https://arrow.apache.org/nanoarrow/latest/reference/python/index.html).
+
+### C/C++
+
+The nanoarrow 0.5.0 release includes a number of bugfixes and improvements
+to the core C library and C++ helpers.
+
+First, the CMake build system was refactored to enable `FetchContent` to
+work in a wider variety of
+[develop/build/install scenarios](https://github.com/apache/arrow-nanoarrow/tree/main/examples/cmake-scenarios). In most cases, CMake-based projects should be able
+to add the nanoarrow C library as a dependency with:
+
+```cmake
+include(FetchContent)
+fetchcontent_declare(nanoarrow
+                     GIT_REPOSITORY https://github.com/apache/arrow-nanoarrow.git
+                     GIT_TAG  apache-arrow-nanoarrow-0.5.0
+                     GIT_SHALLOW TRUE)
+fetchcontent_makeavailable(nanoarrow)
+
+add_executable(some_target ...)
+target_link_libraries(some_target nanoarrow::nanoarrow)
+```
+
+Projects using the [Meson](https://mesonbuild.com/) build system can add a `subprojects/nanoarrow.wrap` file with:
+
+```
+[wrap-git]
+url = https://github.com/apache/arrow-nanoarrow.git
+revision = apache-arrow-nanoarrow-0.5.0
+
+[provide]
+nanoarrow = nanoarrow_dep
+```
+
+...and use `dependencies: [nanoarrow_dep]` to add a dependency:
+
+```
+nanoarrow_dep = dependency('nanoarrow')
+example_exec = executable('some_target',
+                          ...,
+                          dependencies: [nanoarrow_dep])
+```
+
+We are working on adding nanoarrow to the
+[wrapdb](https://mesonbuild.com/Wrapdb-projects.html)
+to make it even easier to add nanoarrow as a dependency in projects that
+use the Meson build system.
```

Comment:
You won't need to create the wrap file yourself. 

```sh
mkdir -p subprojects
meson wrap install nanoarrow
```

will take care of that for you (once we get the wrapdb entry created)

Everything else looks great though

Reply:
Ah sorry misread this a bit. Ultimately up to you if you want to advertise the manually constructed file. It would be more idiomatic from a meson perspective to use the wrapdb, though either will work